tools/shared/src/main/scala/org/scalajs/core/tools/javascript/ScalaJSClassEmitter.scala

```diff
@@ -27,7 +28,8 @@ import org.scalajs.core.tools.linker.{LinkedClass, LinkingUnit}
  */
 final class ScalaJSClassEmitter(
     private[javascript] val outputMode: OutputMode,
-    linkingUnit: LinkingUnit) {
+    linkingUnit: LinkingUnit,
+    private[javascript] val incClassEmitter:  IncClassEmitter) {
```

Comment:
This constructor should be `private[javascript]` now, because only an `IncClassEmitter` should be allowed to create a a `ScalaJSClassEmitter`.